src/portable/synopsys/dwc2/hcd_dwc2.c

```diff
@@ -1000,11 +1000,18 @@ static bool handle_channel_out_slave(dwc2_regs_t* dwc2, uint8_t ch_id, uint32_t
   } else if (hcint & HCINT_ACK) {
     xfer->err_count = 0;
     channel->hcintmsk &= ~HCINT_ACK;
-    if (hcsplt.split_en && !hcsplt.split_compl) {
-      // start split is ACK --> do complete split
-      hcsplt.split_compl = 1;
-      channel->hcsplt = hcsplt.value;
-      channel->hcchar |= HCCHAR_CHENA;
+    if (hcsplt.split_en) {
+      if (!hcsplt.split_compl) {
+        // start split is ACK --> do complete split
+        hcsplt.split_compl = 1;
+        channel->hcsplt = hcsplt.value;
+        channel->hcchar |= HCCHAR_CHENA;
+      }
+    } else {
+      // Device is ready, resume transfer
```

Comment:
we don't need to use channel_xfer_start() since all meta data is setup, also it will toggle PID (causing data toggle error). Just enabled channel with `channel->hcchar |= HCCHAR_CHENA;` is good enough.